kratos/includes/properties_accessor.h

```diff
@@ -0,0 +1,243 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+#if !defined(KRATOS_PROPERTIES_ACCESSOR_H)
+#define  KRATOS_PROPERTIES_ACCESSOR_H
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/properties.h"
+
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+class PropertiesAccessor : public Properties
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of PropertiesAccessor
+    KRATOS_CLASS_POINTER_DEFINITION(PropertiesAccessor);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    PropertiesAccessor(){}
+
+    /// Destructor.
+    virtual ~PropertiesAccessor(){}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    template<class TVariableType>
+    typename TVariableType::Type& operator[](const TVariableType& rV,
+                                             const Geometry::Pointer pGeometry,
+                                             const DataValueContainer& rDataContainer,
+                                             const ProcessInfo& rCurrentProcessInfo,
```

Comment:
ShapeFunctions should be definitely passed and not obtained from the geometry.

this is fundamental for example for immersed elements, where the trick is exactly about modifying the shape functions.
also the gauss_point index should be passed

Reply:
mmm, also i don't think we should overload the operator[] but rather define a proper function.

Reply:
Why not retrieve the SF through the Geometry?
Wouldn't it be quite some overhead to compute them all the time just to pass them, even though they might not be needed?

I agree abt the function, this is just copy-pasted from properties